client/app/scripts/superdesk/editor/spellcheck/spellcheck.js

```diff
@@ -197,31 +173,24 @@ function SpellcheckService($q, api, dictionaries, editor) {
      * Highlight `error` word in node elem
      */
     function hiliteError(node, error) {
-        var regexp = new RegExp('\\b' + escapeRegExp(error) + '\\b', 'im'),
-            index, lastIndex = 0, text = node.textContent,
-            selection = document.getSelection();
-        while ((index = text.search(regexp)) > -1) {
-            var range = document.createRange(),
-                start = findTextNode(node, index + lastIndex),
-                end = findTextNode(node, index + lastIndex + error.length);
-            if (start.node === end.node) {
-                // optimize error hilite when the word is within single text node
-                var replace = start.node.splitText(start.offset),
-                    span = document.createElement('span');
-                span.classList.add(ERROR_CLASS);
-                replace.splitText(end.offset - start.offset);
-                span.textContent = replace.textContent;
-                replace.parentNode.replaceChild(span, replace);
-            } else {
-                range.setStart(start.node, start.offset);
-                range.setEnd(end.node, end.offset);
-                selection.removeAllRanges();
-                selection.addRange(range);
-                document.execCommand('hiliteColor', false, COLOR);
-            }
-
-            lastIndex += index + error.length;
-            text = text.substring(index + error.length);
+        var selection = document.getSelection(),
+            range = document.createRange(),
+            start = findTextNode(node, error.index),
+            end = findTextNode(node, error.index + error.word.length);
+        if (start.node === end.node) {
+            // optimize error hilite when the word is within single text node
+            var replace = start.node.splitText(start.offset),
+                span = document.createElement('span');
+            span.classList.add(ERROR_CLASS);
+            replace.splitText(end.offset - start.offset);
+            span.textContent = replace.textContent;
+            replace.parentNode.replaceChild(span, replace);
+        } else {
+            range.setStart(start.node, start.offset);
+            range.setEnd(end.node, end.offset);
+            selection.removeAllRanges();
+            selection.addRange(range);
+            document.execCommand('hiliteColor', false, COLOR);
```

Comment:
Hmmm... should this DOM changing logic really be contained in a service? We should probably refactor this (not in this PR though).